### PR TITLE
Corrected Docs of `Map.updatedWith`

### PR DESCRIPTION
**Pull request by rs017991**
It seems that the original proposal for this method was intended to take a `PartialFunction`, but when that was refined, the docs were not updated.

### CODE REVIEW

**Comment by som-snytt** [comment]
Same typo, if it is a typo, is cut/paste on original PR to `mutable.Map`, `concurrent.Map`.

I wonder if they used "partial" to mean "takes a Option".

**Comment by rs017991** [comment]
@som-snytt I tracked down the original proposal, and it was indeed written with partials in mind:

https://github.com/scala/scala/pull/6827/files#diff-117dce8d6af394ff848de94139ccb7796b1c0ffe8e4f4b872414ce0f1cb2c5f5R107

**Comment by rs017991** [comment]
@som-snytt I'll go ahead and change those other two functions as well.

Is squash+force push kosher in these parts?

**Comment by SethTisue** [comment]
> Is squash+force push kosher in these parts?

Yes, it's encouraged. We don't squash-merge, so by merge time we want clean, self-contained commits. (Whether you wait until the end of the review process to squash, or do it as you go, is up to you.)

**Comment by rs017991** [comment]
@som-snytt @SethTisue Done.

I also fixed a couple other typos that I stumbled on whilst casting a wider net. I hope you don't mind :)

**Comment by som-snytt** [comment]
I was going to make a self-deprecating grammar nerd joke, but in this case, "partially built" doesn't need a hyphen because it's an adverb ending in "ly"; you might hyphenate "home-built solution", for example.

Their ads drive me nuts, but here's a nice post from grammarly (a grammarly-posted blog, since grammarly is a noun):

https://www.grammarly.com/blog/hyphen/

File: src/library/scala/collection/immutable/HashMap.scala [hunk 1]
```diff
@@ -2228,7 +2228,7 @@ private[immutable] final class HashMapBuilder[K, V] extends ReusableBuilder[(K,
 
   private def isAliased: Boolean = aliased != null
 
-  /** The root node of the partially build hashmap */
+  /** The root node of the partially-built hashmap */
```
Comment (som-snytt) [inline src/library/scala/collection/immutable/HashMap.scala]:
```suggestion
  /** The root node of the partially built hashmap. */
```

File: src/library/scala/collection/immutable/HashSet.scala [hunk 2]
```diff
@@ -1970,7 +1970,7 @@ private[collection] final class HashSetBuilder[A] extends ReusableBuilder[A, Has
 
   private def isAliased: Boolean = aliased != null
 
-  /** The root node of the partially build hashmap */
+  /** The root node of the partially-built hashmap */
```
Comment (som-snytt) [inline src/library/scala/collection/immutable/HashSet.scala]:
```suggestion
  /** The root node of the partially built hashmap. */
```

**Review by som-snytt — COMMENTED** [review]
Seeing if I can apply suggestions for a quickie tweak.

**Comment by som-snytt** [comment]
d'oh except it wants to verify each commit and I did not last-only it. Also I can't short-circuit by squash-merging.

**Comment by som-snytt** [comment]
I also dehyphenated "optionally mapped value". The function takes an "Option-valued parameter".

**Review by som-snytt — APPROVED** [review]
Maybe no-one ever sees the Scaladoc of a private var. I wonder if the IDE displays it in a hover? The important thing is that we'll know.

**Comment by som-snytt** [comment]
From the sports news: "... snap their season-high-matching six-game losing streak."

**Pull request merged**